### DISCUSSION TITLE
Does not collect data properly with checkout

### DISCUSSION THREAD

**Discussion by msroky** [woocommerce/woocommerce#45051] [WooCommerce, Blocks, and Site Editing]
I've tried using various plugins for "Cart Abandonment Recovery" and none of the plugins work while "WooCommerce Blocks Checkout" is on. It doesn't seem to capture a good email field.

I should stop using your checkout page, to be able to use cart abandonment recovery plugins.

Please fix this and solve this problem.

**Comment by senadir** [comment]
Hi again @msroky!

First of all, I'd like to thank you for the constant feedback you're bringing to the project, all of your feedback is well received!

Cart abandonment has to be supported by said plugin, what plugin are you using? To my knowledge, only AutomateWoo is integrated with our new Checkout block and supports the cart abandonment feature.

**Comment by msroky** [comment]
Hi, I tried using this plugin https://wordpress.org/plugins/woo-cart-abandonment-recovery/

**Comment by seagoat86** [comment]
I am having the same problem.

"Mautic WooCommerce Integration" from Makewebbetter is not working with WooCommerce Blocks checkout and neither is "Woo Cart Abandonment Recovery" from CartFlows.

**Comment by gigitux** [comment]
Hi there 👋 
 
Thanks for taking the time to make us aware that Mautic WooCommerce Integration and Woo Cart Abandonment Recovery aren't working with the Checkout and Cart blocks. These blocks are still in early access (via the feature plugin) and it's expected that some existing Woo Extensions that integrate with the current checkout flow in WooCommerce core will not work with the blocks. 
 
We're currently working on adding extensibility for the blocks so extensions have a way to make sure they integrate with the new flows being developed. Once that is done we'll have docs prepared and will be reaching out to Woo Extension developers to help them with getting their extensions integrated.
 
There is already [some documentation available here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/tree/trunk/docs/extensibility) that you could share with the developers of that extension now if you want to get this on their radar!
 
Until then you can keep on top of the [status of the Cart & Checkout Blocks in this documentation](https://docs.woocommerce.com/document/cart-checkout-blocks-support-status/).

**Comment by gigitux** [comment]
We make note of all feedback we receive and we also close issues with the feedback as active issues are used for bugs or feature/enhancements. It does not mean that we don’t note or review the feedback. Closed issues can also still be commented on.

**Comment by msroky** [comment]
I contacted several plugin owners regarding "cart abandonment" plugins, I got answers like this:

"If you do something to change the name or class on your checkout page that is different from the WooCommerce standard checkout page, our plugin will be unable to detect and collect the emails."

They don’t want to customize their plugin to be compatible with your woocommerce blocks plugin.

This problem still exists, and is really very important for my ecommerce business, please solve it as soon as possible. Thanks.

**Comment by nerrad** [comment]
> "If you do something to change the name or class on your checkout page that is different from the WooCommerce standard checkout page, our plugin will be unable to detect and collect the emails."

Hi @msroky 👋🏻 

It's unfortunate you got that response from the developers of extensions you use. However, as noted earlier, we are diligently [providing ways for extension developers to integrate](https://developer.woocommerce.com/2021/11/09/available-extensibility-interfaces-for-the-cart-and-checkout-blocks/). We have also written quite a bit of documentation and are responsive to extension developers that need help or tweaks to various extensibility interfaces with the new checkout but it is up to extension developers to decide whether they will support it or not. We've also opened up a [forum on GitHub discussions](https://github.com/woocommerce/woocommerce-gutenberg-products-block/discussions/categories/q-a-extension-integrations) for extension developers to reach out to our team.

It's possible that the teams you have reached out to are [not aware of the roadmap we are working towards where the Cart and Checkout blocks will eventually become the default checkout flow in WooCommerce core](https://developer.woocommerce.com/2021/08/31/peek-into-the-woocommerce-blocks-roadmap/). From your answer, it appears they may not realize this. Perhaps that is something you would want to share with them? 

For extensions that are listed in the WooCommerce.com marketplace, in 2022 we will be addressing any lack of support for the new checkout flow in various ways as well.

**Comment by msroky** [comment]
Thanks for the reply and support.

Is there any plugin that you can recommend for "Cart Abandonment Recovery", to me that works with your plugin.
